### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by shadeMe**
### Types of change

Docs

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,11 +59,6 @@ spaCy integration for curated transformers is provided by the
 [`spacy-curated-transformers`](https://github.com/explosion/spacy-curated-transformers)
 package.
 
-## ⚠️ Warning: Tech Preview
```
Comment (danieldk) [inline README.md:62]: We should keep this until the release next week. This is the landing page that people will see, even if this is the `main` branch.

File: curated_transformers/generation/falcon.py [hunk 2]
```diff
@@ -13,6 +15,18 @@ class FalconGenerator(DefaultGenerator, FromHFHub):
     Generator for Falcon model variants.
     """
 
+    def __init__(self, tokenizer: Tokenizer, causal_lm: FalconCausalLM):
```
Comment (danieldk) [inline curated_transformers/generation/falcon.py:18]: This is nice for the documentation, but I think we should keep the classes minimal, so no constructor if it's effectively the same as the superclass constructor. If the generated documentation is not clear, we should change the documentation of the superclass constructor.

I don't think people will use the constructor in practice, but instead load it up with a static factory method.

File: curated_transformers/layers/attention.py [hunk 3]
```diff
@@ -148,12 +145,34 @@ def filter_batch_items(self, mask: Tensor) -> "AttentionMask":
...
 
     def dim(self) -> int:
+        """
+        Return the number of dimensions in the mask.
+        """
         return self.bool_mask.dim()
 
     def merge_mask(self, other: "AttentionMask") -> "AttentionMask":
+        """
+        Merge another attention mask with this one.
```
Comment (danieldk) [inline curated_transformers/layers/attention.py:155]:
```suggestion
        Merge this attention mask with another attention mask.
```

File: curated_transformers/layers/attention.py [hunk 4]
```diff
@@ -148,12 +145,34 @@ def filter_batch_items(self, mask: Tensor) -> "AttentionMask":
...
+        """
         return AttentionMask(self.bool_mask.logical_and(other.bool_mask))
 
-    def logit_mask(self, dtype: torch.dtype):
+    def logit_mask(self, dtype: torch.dtype) -> Tensor:
+        """
+        Generate the logit mask for the given ``dtype``.
+
+        Elements of the mask that are ``False`` are set to the
+        minimum value of the ``dtype`` and the rest to zero.
```
Comment (danieldk) [inline curated_transformers/layers/attention.py:169]: We should probably add something saying that when used in softmax, that _adding_ this mask to the logits results in (near-)zero probabilities for the elements that are ``False``.

File: curated_transformers/layers/embeddings.py [hunk 5]
```diff
@@ -56,10 +56,10 @@ def __init__(
 
     def forward(self, input: Tensor) -> Tensor:
         """
-        Returns the positional embedding for the input.
+        Apply sinusoidal embeddings to the input.
```
Comment (danieldk) [inline curated_transformers/layers/embeddings.py:59]: This change is incorrect. As it is now, it only returns the positional embedding or the input.

File: curated_transformers/layers/embeddings.py [hunk 6]
```diff
@@ -56,10 +56,10 @@ def __init__(
...
 
         :param input:
-            Input.
+            Input to apply the sinusoidal embeddings to.
```
Comment (danieldk) [inline curated_transformers/layers/embeddings.py:62]: See above.

File: curated_transformers/layers/feedforward.py [hunk 7]
```diff
@@ -25,15 +25,23 @@ class PointwiseFeedForward(Module):
...
     Gated Linear Units (`Dauphin et al., 2016`_; `Shazeer, 2020`_) are also
-    supported. Gating applies the transformation
-    :math:`(g(xW_g + b_g) * (xW_1 + b_1))W_2 + b_2`, where :math:`W_g` and :math:`b_g`
-    are the affine transformations for the gate.
+    supported. Gating applies the following transformation:
+
+    .. math::
+        (g(xW_g + b_g) * (xW_1 + b_1))W_2 + b_2
+
+    :math:`W_g` and :math:`b_g` are the affine transformations for the gate.
```
Comment (danieldk) [inline curated_transformers/layers/feedforward.py:44]:
```suggestion
    :math:`W_g` and :math:`b_g` are the affine transformation for the gate.
```

(Also wrong in the original doc.)

File: curated_transformers/models/falcon/layer.py [hunk 8]
```diff
@@ -118,7 +118,7 @@ def forward(
             corresponding mask element is set to ``False`` are ignored
             during attention calculation.
 
-            *Shape:* ``(batch_size, seq_len)``
+            *Shape:* ``(batch_size, 1, 1, seq_len)``
```
Comment (danieldk) [inline curated_transformers/models/falcon/layer.py:121]: Maybe we should consider removing the shape from attention mask arguments? I think the shape makes sense for `Tensor` arguments. But for `AttentionMask` the shape is already checked/enforced by the class itself. Having the shape in the argument doc suggest that this is something that the user should take care of.

File: curated_transformers/tokenizers/legacy/sentencepiece_tokenizer.py [hunk 9]
```diff
@@ -20,7 +20,7 @@ def __init__(
         processor: SentencePieceProcessor,
     ):
         """
-        Construct a sentence piece tokenzier.
+        Construct a tokenzier from ``curated-tokenizers`` SentencePiece processor.
```
Comment (danieldk) [inline curated_transformers/tokenizers/legacy/sentencepiece_tokenizer.py:23]:
```suggestion
        Construct a tokenizer from ``curated-tokenizers`` SentencePiece processor.
```

File: curated_transformers/tokenizers/tokenizer.py [hunk 10]
```diff
@@ -74,7 +74,7 @@ def padded_tensor(
         device: Optional[torch.device] = None,
     ) -> Tensor:
         """
-        Padded CPU tensor of the piece identifiers.
+        Generate a padded tensor of the piece identifiers.
```
Comment (danieldk) [inline curated_transformers/tokenizers/tokenizer.py:77]: Maybe we should add a note here that the actual padding id typically doesn't matter when an attention mask is used, as long as the id is a valid vocabulary index? As a user I would be worried that I have to give some padding identifier, but how do I know which one?

On that note, we could consider just giving it a default value of `0` as well.

File: docs/source/api.rst [hunk 11]
```diff
@@ -1,12 +1,8 @@
 API
 ===
 
-.. attention::
-   This library is currently in the tech preview stage - it is possible that models 
-   and their APIs will change in incompatible ways before ``1.0.0``.
-
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
```
Comment (danieldk) [inline docs/source/api.rst:5]: Too messy. A ToC should provide a quick overview of the structure of the documentation, 3 is very overwhelming.

File: docs/source/conf.py [hunk 12]
```diff
@@ -38,7 +38,7 @@
     # Toc options
     "collapse_navigation": False,
     "sticky_navigation": True,
-    "navigation_depth": 3,
+    "navigation_depth": 4,
```
Comment (danieldk) [inline docs/source/conf.py:41]: In contrast to above, I think 4 is fine here. I think it's only used for the sidebar (?) which uses progressive disclosure.

File: curated_transformers/generation/falcon.py (hunk 2 above)
Comment (shadeMe) [reply curated_transformers/generation/falcon.py:18]: >  If the generated documentation is not clear, we should change the documentation of the superclass constructor.

Do you have a suggestion? If we don't override the ctor here, the docstring of the `GeneratorWrapper` ctor will be displayed, but that results in an inconsistency b'ween the Falcon and Dolly v2 generators - The latter doesn't allow a default generation config to be passed to the ctor but the former will. Same with the Llama generator.

File: curated_transformers/layers/embeddings.py (hunk 6 above)
Comment (shadeMe) [reply curated_transformers/layers/embeddings.py]: Reverted.

File: curated_transformers/layers/embeddings.py (hunk 5 above)
Comment (shadeMe) [reply curated_transformers/layers/embeddings.py]: Indeed! reverted.

File: README.md (hunk 1 above)
Comment (shadeMe) [reply README.md]: Reverted.

File: curated_transformers/models/falcon/layer.py (hunk 8 above)
Comment (shadeMe) [reply curated_transformers/models/falcon/layer.py]: Fair point. Removed.

File: docs/source/api.rst (hunk 11 above)
Comment (shadeMe) [reply docs/source/api.rst]: Done.

File: curated_transformers/tokenizers/tokenizer.py (hunk 10 above)
Comment (shadeMe) [reply curated_transformers/tokenizers/tokenizer.py:79]: Done.

File: curated_transformers/generation/falcon.py (hunk 2 above)
Comment (danieldk) [reply curated_transformers/generation/falcon.py:18]: Good point, let's stick with this.

**Review by danieldk — APPROVED** [review]
LGTM, there are still some merge conflicts that need to be resolved.
